inc/auth.class.php

```diff
@@ -581,12 +581,16 @@ function Login($login_name, $login_password, $noauto=false) {
          if ($this->getAlternateAuthSystemsUserLogin($authtype)
              && !empty($this->user->fields['name'])) {
             // Used for log when login process failed
-            $login_name                     = $this->user->fields['name'];
-            $this->auth_succeded            = true;
-            $this->extauth                  = 1;
-            $this->user_present             = $this->user->getFromDBbyName(addslashes($login_name));
-            $this->user->fields['authtype'] = $authtype;
-            $user_dn                        = false;
+            $login_name                        = $this->user->fields['name'];
+            $this->auth_succeded               = true;
+            $this->user_present                = $this->user->getFromDBbyName(addslashes($login_name));
+            if (self::isAlternateAuth($authtype)) {
```

Comment:
question for this part:
   do we need to condition this assignment : "$this->user->fields['authtype'] = $authtype;" ?

i think these lines (587-592) could be reduced to:
```php
$this->user->fields['authtype'] = $authtype;
$this->extauth = self::isAlternateAuth($authtype);
```

without any aditionnal condition

More, futher in the code, the assignment of authtype and extauth are done again, maybe we could remove next part.